### DISCUSSION TITLE
PostmarketOS kernel compilation issue

### DISCUSSION THREAD

**Discussion by mj-sakellaropoulos** [hansemro/kc1-linux#33] [Q&A] [answered]
I've tried to get the kernel compiling (in their chroot environment), but this is what I get :
```
In file included from ./../include/linux/compiler_types.h:85,
                 from <command-line>:
../include/linux/compiler-gcc.h:2:2: error: #error "Please don't include <linux/compiler-gcc.h> directly, include <linux/compiler.h> instead."
    2 | #error "Please don't include <linux/compiler-gcc.h> directly, include <linux/compiler.h> instead."
```

using latest armv7 toolchain (afaik)

**Comment by hansemro** [comment]
`pmbootstrap` is broken on my end. Last time I tried, it could not compile mainline.

**Reply by mj-sakellaropoulos** [reply]
I removed the various patches and it got further, it looks like their toolchain/gcc has some problems

**Comment by hansemro** [comment, marked as answer]
`pmbootstrap log` should have shown you this:
```
If your build breaks with 'Please
don't include <linux/compiler-gcc.h> directly' or a similar
compiler-gcc.h related error, then set
  REPLACE_GCCH=0
in your kernel APKBUILD at the start of the
downstreamkernel_prepare.sh line.
```

Adding that flag fixes the issue for me.

so:
```
 prepare() {
     default_prepare
     REPLACE_GCCH=0 . downstreamkernel_prepare
 }
```

**Reply by hansemro** [reply]
<details>
<summary>build success</summary>
<pre>
  LD [M]  drivers/usb/dwc2/dwc2.ko
  LD [M]  drivers/usb/host/ehci-omap.ko
  LD [M]  drivers/usb/phy/phy-am335x-control.ko
  LD [M]  drivers/usb/phy/phy-am335x.ko
  LD [M]  drivers/usb/phy/phy-generic.ko
  LD [M]  drivers/usb/phy/phy-twl6030-usb.ko
  LD [M]  drivers/usb/roles/roles.ko
  LD [M]  fs/autofs/autofs4.ko
  LD [M]  lib/crypto/libarc4.ko
  LD [M]  lib/crypto/libchacha.ko
  LD [M]  lib/crypto/libsha256.ko
  LD [M]  net/mac80211/mac80211.ko
  LD [M]  net/phonet/pn_pep.ko
  LD [M]  net/phonet/phonet.ko
  LD [M]  net/wireless/cfg80211.ko
make[1]: Leaving directory '/home/pmos/build/src/linux-f40ddce88593482919761f74910f42f4b84c004b/out'
>>> linux-amazon-otter: Entering fakeroot...
zImage found: zImage
>>> linux-amazon-otter*: Running postcheck for linux-amazon-otter
>>> linux-amazon-otter*: Preparing package linux-amazon-otter...
>>> WARNING: linux-amazon-otter*: No arch specific binaries found so arch should probably be set to "noarch"
>>> linux-amazon-otter*: Tracing dependencies...
>>> linux-amazon-otter*: Package size: 7.5 MB
>>> linux-amazon-otter*: Compressing data...
>>> linux-amazon-otter*: Create checksum...
>>> linux-amazon-otter*: Create linux-amazon-otter-5.11.0-r0.apk
>>> linux-amazon-otter: Build complete at Thu, 06 May 2021 15:20:43 +0000 elapsed time 0h 7m 23s
>>> linux-amazon-otter: Updating the pmos/armv7 repository index...
>>> linux-amazon-otter: Signing the index...
(2815329) [08:20:43] NOTE: chroot is still active (use 'pmbootstrap shutdown' as necessary)
(2815329) [08:20:43] Done
</pre>
</details>

**Reply by mj-sakellaropoulos** [reply]
Nice!

I could have sworn I saw that string somewhere I just couldn't remember where it was!! I searched google for a while hahaha
